File: RecipeSharingMenu/backend/src/models/Comment.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Comment = sequelize.define('Comment', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  user_id: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  recipe_id: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  content: {
    type: DataTypes.TEXT,
    allowNull: false
  },
  parent_id: {
    type: DataTypes.INTEGER,
    defaultValue: null,
    comment: '回复的评论ID'
  }
}, {
  tableName: 'comments'
});

module.exports = Comment;

File: OnlineKnowledgePaid/backend/model/comment.go
package model

import "time"

type Comment struct {
	ID        uint64    `gorm:"primaryKey;autoIncrement" json:"id"`
	ArticleID uint64    `gorm:"not null" json:"article_id"`
	UserID    uint64    `gorm:"not null" json:"user_id"`
	Content   string    `gorm:"type:text;not null" json:"content"`
	ParentID  *uint64   `json:"parent_id"`
	LikeCount int       `gorm:"not null;default:0" json:"like_count"`
	Status    int8      `gorm:"not null;default:1" json:"status"`
	CreatedAt time.Time `gorm:"autoCreateTime" json:"created_at"`
	User      *User     `gorm:"foreignKey:UserID" json:"user,omitempty"`
	Article   *Article  `gorm:"foreignKey:ArticleID" json:"article,omitempty"`
}

func (Comment) TableName() string {
	return "comments"
}
